src/org/joni/Config.java

```diff
@@ -22,54 +22,55 @@
+    final boolean USE_NEWLINE_AT_END_OF_STRING_HAS_EMPTY_LINE = ConfigSupport.getBoolean("joni.use_newline_at_end_of_string_has_empty_line", true); /* /\n$/ =~ "\n" */
+    final boolean USE_WARNING_REDUNDANT_NESTED_REPEAT_OPERATOR = ConfigSupport.getBoolean("joni.use_warning_redundant_nested_repeat_operator", true);
+
+    final boolean CASE_FOLD_IS_APPLIED_INSIDE_NEGATIVE_CCLASS = ConfigSupport.getBoolean("joni.case_fold_is_applied_inside_negative_cclass", true);
+
+    final boolean USE_MATCH_RANGE_MUST_BE_INSIDE_OF_SPECIFIED_RANGE = ConfigSupport.getBoolean("joni.use_match_range_must_be_inside_of_specified_range", false);
+    final boolean USE_CAPTURE_HISTORY = ConfigSupport.getBoolean("joni.use_capture_history", false);
+    final boolean USE_VARIABLE_META_CHARS = ConfigSupport.getBoolean("joni.use_variable_meta_chars", true);
+    final boolean USE_WORD_BEGIN_END = ConfigSupport.getBoolean("joni.use_word_begin_end", true); /* "\<": word-begin, "\>": word-end */
+    final boolean USE_FIND_LONGEST_SEARCH_ALL_OF_RANGE = ConfigSupport.getBoolean("joni.use_find_longest_search_all_of_range", true);
+    final boolean USE_SUNDAY_QUICK_SEARCH = ConfigSupport.getBoolean("joni.use_sunday_quick_search", true);
+    final boolean USE_CEC = ConfigSupport.getBoolean("joni.use_cec", false);
+    final boolean USE_DYNAMIC_OPTION = ConfigSupport.getBoolean("joni.use_dynamic_option", false);
+    final boolean USE_BYTE_MAP = ConfigSupport.getBoolean("joni.use_byte_map", OptExactInfo.OPT_EXACT_MAXLEN <= CHAR_TABLE_SIZE);
+    final boolean USE_INT_MAP_BACKWARD = ConfigSupport.getBoolean("joni.use_int_map_backward", false);
+
+    final int NREGION                   = ConfigSupport.getInt("joni.nregion", 10);
+    final int MAX_BACKREF_NUM           = ConfigSupport.getInt("joni.max_backref_num", 1000);
+    final int MAX_CAPTURE_GROUP_NUM     = ConfigSupport.getInt("joni.max_capture_group_num", 32767);
+    final int MAX_REPEAT_NUM            = ConfigSupport.getInt("joni.max_multi_byte_ranges_num", 100000);
+    final int MAX_MULTI_BYTE_RANGES_NUM = ConfigSupport.getInt("joni.max_multi_byte_ranges_num", 10000);
 
     // internal config
-    final boolean USE_OP_PUSH_OR_JUMP_EXACT         = true;
-    final boolean USE_QTFR_PEEK_NEXT                = true;
+    final boolean USE_OP_PUSH_OR_JUMP_EXACT         = ConfigSupport.getBoolean("joni.use_op_push_or_jump_exact", true);
+    final boolean USE_QTFR_PEEK_NEXT                = ConfigSupport.getBoolean("joni.use_qtfr_peek_next", true);
 
-    final int INIT_MATCH_STACK_SIZE                 = 64;
+    final int INIT_MATCH_STACK_SIZE                 = ConfigSupport.getInt("joni.init_match_stack_size", 64);
 
-    final boolean DONT_OPTIMIZE                     = false;
+    final boolean DONT_OPTIMIZE                     = ConfigSupport.getBoolean("joni.dont_optimize", false);
```

Comment:
Let's just make this OPTIMIZE/joni.optimize=true. The double negative here makes my head hurt.

Reply:
I'm assuming the latter - pushed that, happy to adjust otherwise

Reply:
Yeah good call on preserving the public API. We might as well deprecate it too. I doubt anyone is using it.

Reply:
👍 deprecated it